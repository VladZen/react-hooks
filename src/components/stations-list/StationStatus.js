import Enum from '@/types/PropTypeEnum'

import Icon from '@/components/Icon'

const StationStatus = ({ available }) => {
  const status = available
    ? 'available'
    : 'offline'

  return (
    <span className='stations__item__status stations__item__status--offline'>
      <Icon
        alias={ status }
        className='stations__item__status__icon'
      />

      <span className='text-capitalize'>
        { status }
      </span>
    </span>
  )
}

StationStatus.propTypes = {
  available: Enum()
}

export default StationStatus
