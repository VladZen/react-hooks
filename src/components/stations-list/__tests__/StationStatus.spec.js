import { unmountComponentAtNode, render } from 'react-dom'
import { act } from 'react-dom/test-utils'
import StationStatus from '@/components/stations-list/StationStatus'

let container = null
beforeEach(() => {
  container = document.createElement('div')
  document.body.appendChild(container)
})

afterEach(() => {
  unmountComponentAtNode(container)
  container.remove()
  container = null
})

describe('StationStatus component', () => {
  describe('rendering', () => {
    it('available', () => {
      act(() => {
        render(
          <StationStatus available={ 1 } />,
          container
        )
      })

      expect(container).toMatchSnapshot()
    })

    it('offline', () => {
      act(() => {
        render(
          <StationStatus available={ 0 } />,
          container
        )
      })

      expect(container).toMatchSnapshot()
    })
  })
})
