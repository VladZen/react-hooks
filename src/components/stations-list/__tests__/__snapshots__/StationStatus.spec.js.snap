// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StationStatus component rendering available 1`] = `
<div>
  <span
    class="stations__item__status stations__item__status--offline"
  >
    <svg
      class="stations__item__status__icon"
    >
      <use
        href="sprite.svg#available"
      />
    </svg>
    <span
      class="text-capitalize"
    >
      available
    </span>
  </span>
</div>
`;

exports[`StationStatus component rendering offline 1`] = `
<div>
  <span
    class="stations__item__status stations__item__status--offline"
  >
    <svg
      class="stations__item__status__icon"
    >
      <use
        href="sprite.svg#offline"
      />
    </svg>
    <span
      class="text-capitalize"
    >
      offline
    </span>
  </span>
</div>
`;
